### ISSUE TITLE
Layer support

### ISSUE THREAD

**Issue by rdrpenguin04** [LightningCreations/libpipeworks#4]
Goal: To implement a "Layer" construct that organizes Things into a group that renders with the same depth.

Constraints:
- Constructing a Layer should return something usable as a Thing.
- Layer should have a function to add/remove Things.
- Layer should have a function to either:
  - Translate the layer rendering
  - Translate the Things on the layer (only in effect; any x/y attributes of the Thing in question should remain constant)
- A Layer should render as if all of its Things were rendered, in depth sorted order, followed by being compressed into one graphic, then rendered to the screen at the Layer's depth.

**Comment by chorman0773** [comment]
This seems similar to the DrawingSpace api from JPipeworksEngine <https://github.com/LightningCreations/JPipeworksEngine/pull/1>. 
Would it make sense to add additional apis to Layer's similar to that (Affine Transformations, Colour Mapping by function, etc.), in addition to simple translations?

**Comment by rdrpenguin04** [comment]
Adding those at some point is probably a good idea; adding them now
probably isn't. DrawingSpace was too ambitious a goal, and it showed.

Those APIs could be put into a ComplexLayer; that would also be an
optimization opportunity, since only the Layers that needed the additional
processing would need to take it.

My suggestion would be to start a new issue with your ideas, and we'll work
on that once the base engine is done.
